src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.Windows.cs

```diff
@@ -67,8 +65,45 @@ public static byte[] ConvertAlpnProtocolListToByteArray(List<SslApplicationProto
 
             if (context == null && sslAuthenticationOptions.ApplicationProtocols != null && sslAuthenticationOptions.ApplicationProtocols.Count != 0)
             {
-                byte[] alpnBytes = ConvertAlpnProtocolListToByteArray(sslAuthenticationOptions.ApplicationProtocols);
-                inputBuffers.SetNextBuffer(new InputSecurityBuffer(new ReadOnlySpan<byte>(alpnBytes), SecurityBufferType.SECBUFFER_APPLICATION_PROTOCOLS));
+                var alpn = sslAuthenticationOptions.ApplicationProtocols;
+
+                if (alpn.Count == 1 && alpn[0] == SslApplicationProtocol.Http11)
+                {
+                    inputBuffers.SetNextBuffer(new InputSecurityBuffer(s_http1, SecurityBufferType.SECBUFFER_APPLICATION_PROTOCOLS));
+                }
+                else if (alpn.Count == 1 && alpn[0] == SslApplicationProtocol.Http2)
+                {
+                    inputBuffers.SetNextBuffer(new InputSecurityBuffer(s_http2, SecurityBufferType.SECBUFFER_APPLICATION_PROTOCOLS));
+                }
+                else if (alpn.Count == 2 && alpn[0] == SslApplicationProtocol.Http11 && alpn[1] == SslApplicationProtocol.Http2)
+                {
+                    inputBuffers.SetNextBuffer(new InputSecurityBuffer(s_http12, SecurityBufferType.SECBUFFER_APPLICATION_PROTOCOLS));
+
+                }
+                else if (alpn.Count == 2 && alpn[0] == SslApplicationProtocol.Http2 && alpn[1] == SslApplicationProtocol.Http11)
+                {
+                    inputBuffers.SetNextBuffer(new InputSecurityBuffer(s_http21, SecurityBufferType.SECBUFFER_APPLICATION_PROTOCOLS));
+                }
+                else
+                {
+                    int protocolLenngth = Interop.Sec_Application_Protocols.GetProtocolLength(alpn);
+                    int bufferLength = sizeof(Interop.Sec_Application_Protocols) + protocolLenngth;
+
+                    Span<byte> alpnBuffer;
+                    if (bufferLength <= 64)
+                    {
+                        // workaround for CS8350
```

Comment:
If you declare the local as `scoped Span<byte> alpnBuffer`, you should not need this workaround.

Reply:
nice. updated.